spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroReader.java

```diff
@@ -39,34 +41,51 @@ protected void writeAndValidate(Schema schema) throws IOException {
   }
 
   @Override
-  protected void writeAndValidate(Schema writeSchema, Schema expectedSchema) throws IOException {
-    List<Record> expected = RandomData.generateList(writeSchema, 100, 0L);
-
-    OutputFile outputFile = new InMemoryOutputFile();
```

Comment:
This backport wasn't exactly clean because 3.5 creates the test files a bit differently (and so there's newer imports as well adding to the diff) but it's still effecitvely the same.  I just took the 3.5 ones as is just to be consistent